pkg/component/etcd/etcd.go

```diff
@@ -693,12 +693,6 @@ func (e *etcd) RolloutPeerCA(ctx context.Context) error {
 	}
 
 	_, err := controllerutils.GetAndCreateOrMergePatch(ctx, e.client, e.etcd, func() error {
-		// Exit early if etcd object has already the expected CA reference.
```

Comment:
Do you remember the reason why we had added this in the first place?

Reply:
Good question. IIRC, this was an optimization to avoid unnecessary reconciliations during the `preparing` step if the flow/reconciliation starts again.